packages/coreutils/src/statedb.ts

```diff
@@ -12,8 +12,9 @@ import { IStateDB } from './tokens';
 /**
  * The default concrete implementation of a state database.
  */
-export class StateDB<T extends ReadonlyJSONValue = ReadonlyJSONValue>
-  implements IStateDB<T> {
+export class StateDB<
+  T extends ReadonlyPartialJSONValue = ReadonlyPartialJSONValue
+> implements IStateDB<T> {
```

Comment:
Here I switch the StateDB to use partial types as well, as that seems to be needed / more accurate. On the other hand, it is harder to tighten up this types later if we incorrectly relax them now. Thoughts? CC @afshin